bundles/org.openhab.binding.intellicenter2/src/main/java/org/openhab/binding/intellicenter2/internal/handler/IntelliCenter2ThingHandler.java

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.intellicenter2.internal.handler;
+
+import static org.openhab.binding.intellicenter2.internal.protocol.Attribute.OBJNAM;
+
+import java.util.Objects;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.intellicenter2.internal.model.ResponseModel;
+import org.openhab.binding.intellicenter2.internal.protocol.Attribute;
+import org.openhab.binding.intellicenter2.internal.protocol.ICProtocol;
+import org.openhab.binding.intellicenter2.internal.protocol.NotifyListListener;
+import org.openhab.binding.intellicenter2.internal.protocol.ResponseObject;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Abstract Handler that provides a consistent way to support other IntelliCenter2 handlers.
+ *
+ * @author Valdis Rigdon - Initial contribution
+ */
+@NonNullByDefault
+public abstract class IntelliCenter2ThingHandler<T extends ResponseModel> extends BaseThingHandler
+        implements NotifyListListener {
+
+    private final Logger logger = LoggerFactory.getLogger(IntelliCenter2ThingHandler.class);
+
+    public IntelliCenter2ThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        scheduler.execute(() -> {
+            try {
+                // get an updated model from IC
+                final ICProtocol protocol = getProtocol();
+                if (protocol == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR);
+                    return;
+                }
+                final T model = queryModel(protocol);
+                getProtocol().subscribe(this, model.asRequestObject());
+                updateStatus(ThingStatus.ONLINE);
+                updateState(model);
+            } catch (Exception e) {
+                logger.error("Error refreshing model", e);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
```

Comment:
Currently it seems this will actually stop the binding from working. If this is actually a communication error which can be caused by temporary network outage (as example), it would probably be better to schedule a job so this would be retried?